src/activities/Elsa.Activities.Email/Activities/SendEmail/SendEmail.cs

```diff
@@ -94,9 +100,19 @@ protected override async ValueTask<IActivityExecutionResult> OnExecuteAsync(Acti
             SetRecipientsEmailAddresses(message.Cc, Cc);
             SetRecipientsEmailAddresses(message.Bcc, Bcc);
 
-            await _smtpService.SendAsync(context, message, context.CancellationToken);
+            var outcomes = new List<string> { OutcomeNames.Done };
+            try
+            {
+                await _smtpService.SendAsync(context, message, context.CancellationToken);
+                outcomes.Add("Success");
+            }
+            catch (Exception ex)
+            {
+                outcomes.Add("Unexpected Error");
+                outcomes.Add(ex.Message);
```

Comment:
Adding `ex.Message` as an outcome allows the user to take action based on a specific error message, which might be useful. But if the message is really long, that will make for a really long outcome name as well. Better might be to use the exception type name and/or an error code. But then we also need to allow the user to specify what errors they want to handle so that the designer can display the supported errors.

We could do it if someone has an actual need for it, but if that's not currently the case I suggest we just add the error message as a data item, like this:

```csharp
context.JournalData.Add("Error", ex.Message);
```